### PR TITLE
Handle Perl and R matrix items.

### PR DESCRIPTION
**Pull request by pelson**
Again, unittests would be great, but I have tested this by hand to confirm that this now enables Perl and R support to feedstocks.

### CODE REVIEW

**Comment by pelson** [comment]
Closes #82 by simply removing the appveyor.yml if it isn't needed.

**Comment by jakirkham** [comment]
Nice, so we still register for AppVeyor, but simply don't include the CI file. Awesome fix. :+1:

**Comment by jakirkham** [comment]
So, we tried removal of `appveyor.yml` on [autoconf]( https://github.com/conda-forge/autoconf-feedstock/pull/2 ), but it doesn't seem to quite work because we might need to change a setting on AppVeyor. I don't know if there is a way for you to do that, as well. If not, another strategy might be to add a skip commit filter that [skips all commits]( https://github.com/conda-forge/autoconf-feedstock/pull/3 ). This won't avoid AppVeyor jobs getting queued, but it should at least allow for a fast finish.

**Comment by pelson** [comment]
> Just out of curiosity, under what conditions does it get added back?

If your recipe no longer skips the specific windows matrix item, rerendering with conda-smithy (which anybody can do) will restore the ```appveyor.yml```.

**Comment by pelson** [comment]
> So, we tried removal of appveyor.yml on autoconf, but it doesn't seem to quite work because we might need to change a setting on AppVeyor. I don't know if there is a way for you to do that, as well.

There is a toggle which states ```skip if no appveyor.yml``` in AppVeyor itself which needs to be enabled. By default it is not, and so I will have to toggle that for all feedstocks (as well as all future feedstocks).

**Comment by jakirkham** [comment]
> rerendering with conda-smithy (which anybody can do) will restore the `appveyor.yml`.

Got it. That's what I was thinking. Thanks for clarifying.

> By default it is not, and so I will have to toggle that for all feedstocks (as well as all future feedstocks).

Right, what would we need to do in order to get that to work? Is it possible to automate?

**Comment by jakirkham** [comment]
Are you just planning to use [this API]( https://www.appveyor.com/docs/api/projects-builds#update-project )?

**Comment by pelson** [comment]
> Are you just planning to use this API?

Yep

**Comment by jakirkham** [comment]
So, I'm thinking about this again and I think we should just rename the CI file instead of deleting it so it is easy to get back. Mainly I am thinking about this in terms of PRs that want to add Windows support to something that didn't have it. For instance, we could move `appveyor.yml` to `disabled_appveyor.yml`. When someone wants to add support back, we just tell the to rename back to `appveyor.yml`. I will draw up a PR and we can discuss.

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -76,18 +99,26 @@ def render_appveyor(jinja_env, forge_config, forge_dir):
...
+                cases_not_skipped.append(case)
+        matrix = cases_not_skipped
+
+    target_fname = os.path.join(forge_dir, 'appveyor.yml')
+
+    if not matrix:
+        # There are no cases to build (not even a case without any special
+        # dependencies), so remove the appveyor.yml if it exists.
+        if os.path.exists(target_fname):
+            os.remove(target_fname)
```
Comment (jakirkham) [inline conda_smithy/configure_feedstock.py]: It will probably just be changing this line honestly.

**Comment by jakirkham** [comment]
Proposal to rename AppVeyor CI files here ( https://github.com/conda-forge/conda-smithy/pull/94 ).

**Comment by pelson** [comment]
> When someone wants to add support back, we just tell the to rename back to appveyor.yml

Not sure I agree with this statement (though I agree with your change in #94). ``conda-smithy`` shouldn't be a magical black box, and anybody should be able to:

```
conda install -c conda-forge conda-smithy
conda smithy rerender
```

There are other projects out there who demand this kind of thing from their users, and expect the rendered content to be considered a black box (e.g. versioneer).

Anyway, agree or not, the change you put forwards in #94 was a nice one :+1:

**Pull request merged**